crates/driver/src/infra/persistence/mod.rs

```diff
@@ -0,0 +1,133 @@
+use {
+    crate::{
+        domain::liquidity::{self},
+        infra::{api::Auction, solver::Config},
+    },
+    number::serialization::HexOrDecimalU256,
+    primitive_types::U256,
+    serde::{Deserialize, Serialize},
+    serde_with::serde_as,
+    std::sync::Arc,
+    tracing::Instrument,
+};
+
+#[serde_as]
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct AuctionWithLiquidity {
+    auction: Auction,
+    liquidity: Vec<Liquidity>,
+}
+
+#[serde_as]
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Liquidity {
```

Comment:
There is no need to add any new DTOs or serialize implementations. IIUC you could simply upload the file [here](https://github.com/cowprotocol/services/blob/main/crates/driver/src/infra/solver/mod.rs#L189) without having to worry about any serialization because the request body was already serialized. And since you have access to the `solver` there you should be able to call the respective upload function from there.

Reply:
@MartinquaXD I understand, thanks for pointing it out. I changed it so now the body is updated to the S3, which contains the desired auction + liquidity + some small extra configuration which may be nice to have.